packages/actor-query-operation-extend/lib/ActorQueryOperationExtend.ts

```diff
@@ -33,7 +33,7 @@ export class ActorQueryOperationExtend extends ActorQueryOperationTypedMediated<
     );
 
     const extendKey = termToString(variable);
-    const config = { ...ActorQueryOperation.getExpressionContext(context, this.mediatorQueryOperation) };
+    const config = { ...ActorQueryOperation.getAsyncExpressionContext(context, this.mediatorQueryOperation) };
```

Comment:
Is there a certain reason why this was changed?

We probably want the async context indeed, but I'm just wondering why it wasn't async in the first place.

Reply:
sparqlee's typings specify two different types of contexts, one the sync evaluator and another one for the async evaluator. The former one expects synchronous `bnode()` and `exists()` functions while the latter one expects asynchronous versions of those functions. The `exists()` function provided by comunica is always asynchronous and thus can only be included in the asynchronous context, hence I tried to use the asynchronous context as much as possible so as not to introduce breaking changes.

Reply:
I think Comunica only makes use of the async context, or am I wrong there? If not, perhaps we can just have `getExpressionContext` always return an async context?

Reply:
That's what I thought, too, but then I found https://github.com/comunica/comunica/blob/7e77c0134b8354eb735ab2b805f8f6689f7c6b34/packages/actor-query-operation-group/lib/GroupsState.ts#L32  which uses the `SyncEvaluatorConfig` type.

Reply:
Ah yes, aggregation happens sync. Then I guess there's no other way around this :-)